sw/device/silicon_creator/lib/cert/cert.c

```diff
@@ -0,0 +1,220 @@
+      .otp_creator_sw_cfg_hash_size = 0,
+      .otp_owner_sw_cfg_hash = NULL,
+      .otp_owner_sw_cfg_hash_size = 0,
+      .otp_hw_cfg0_hash = NULL,
+      .otp_hw_cfg0_hash_size = 0,
+      .debug_flag = is_debug_exposed(),
+      .creator_pub_key_id = (unsigned char *)uds_pubkey_id->digest,
+      .creator_pub_key_id_size = kCertKeyIdSizeInBytes,
+      .auth_key_key_id = perso_data_in->auth_key_key_id,
+      .auth_key_key_id_size = kCertKeyIdSizeInBytes,
+      .creator_pub_key_ec_x = (unsigned char *)curr_pubkey.x,
+      .creator_pub_key_ec_x_size = kAttestationPublicKeyCoordBytes,
+      .creator_pub_key_ec_y = (unsigned char *)curr_pubkey.y,
+      .creator_pub_key_ec_y_size = kAttestationPublicKeyCoordBytes,
+  };
+  TRY(uds_build_tbs(&uds_cert_tbs_params, tbs_cert, tbs_cert_size));
+
+  return OK_STATUS();
+}
+
+status_t gen_cdi_0_keys_and_cert(manuf_cert_perso_data_in_t *perso_data_in,
+                                 hmac_digest_t *uds_pubkey_id,
+                                 hmac_digest_t *cdi_0_pubkey_id, uint8_t *cert,
+                                 size_t *cert_size) {
+  TRY(keymgr_state_check(kKeymgrStateCreatorRootKey));
+
+  // Set attestation binding to the ROM_EXT / Ownership Manifest measurements.
+  hmac_digest_t combined_measurements;
+  uint32_t rom_ext_measurements[kAttestMeasurementSizeIn32BitWords * 2];
+  memcpy(rom_ext_measurements, perso_data_in->rom_ext_measurement,
+         kAttestMeasurementSizeInBytes);
+  memcpy(&rom_ext_measurements[kAttestMeasurementSizeIn32BitWords],
+         perso_data_in->owner_manifest_measurement,
+         kAttestMeasurementSizeInBytes);
+  hmac_sha256(rom_ext_measurements, kAttestMeasurementSizeInBytes * 2,
+              &combined_measurements);
+  memcpy(attestation_binding_value.data, combined_measurements.digest,
+         kAttestMeasurementSizeInBytes);
+  // We set the sealing binding value to all zeros as it is unused in the
+  // current personalization flow. This may be changed in the future.
+  memset(sealing_binding_value.data, 0, kAttestMeasurementSizeInBytes);
+  keymgr_sw_binding_unlock_wait();
+  keymgr_sw_binding_set(&sealing_binding_value, &attestation_binding_value);
+
+  // Generate the CDI_0 key.
+  keymgr_advance_state();
+  TRY(keymgr_state_check(kKeymgrStateOwnerIntermediateKey));
+  TRY(otbn_boot_attestation_keygen(kCdi0AttestationKeySeed,
+                                   kCdi0KeymgrDiversifier, &curr_pubkey));
+
+  // Generate the key ID.
+  hmac_sha256(&curr_pubkey, kAttestationPublicKeyCoordBytes * 2,
+              cdi_0_pubkey_id);
+
+  // Generate the TBS certificate.
+  cdi_0_tbs_values_t cdi_0_cert_tbs_params = {
+      .rom_ext_hash = (unsigned char *)perso_data_in->rom_ext_measurement,
+      .rom_ext_hash_size = kAttestMeasurementSizeInBytes,
+      .owner_manifest_hash =
+          (unsigned char *)perso_data_in->owner_manifest_measurement,
+      .owner_manifest_hash_size = kAttestMeasurementSizeInBytes,
```

Comment:
@cfrantz would it make sense to capture any ownership blob measurements in this stage?  Maybe that is something we can track in a separate issue.

Reply:
Ah, yes, after speaking with @cfrantz , I was going to move this measurement to the CDI_1 stage). I was planning to do that in a follow up if that is ok? since this is already quite a large PR.

Reply:
Added in #21539.